docs/homepage/blog/offline_reinforcement_learning_algorithm_phase1/index.md

```diff
@@ -0,0 +1,315 @@
+    pre::P = identity
+    μ::U
+    logσ::S
+    min_σ::Float32 = 0f0
+    max_σ::Float32 = Inf32
+end
+```
+We have improved the evaluation function and inference function of `GaussianNetwork`. By evaluation function, given the state, then the mean and log-standard deviation are obtained. Furthermore, we can sample the action from distribution and get the probability of the action in a given state. When calling the inference function with parameter state and action, we get the likelihood of the action in a given state.
+
+```julia
+### Evaluation
+function (model::GaussianNetwork)(state; is_sampling::Bool=false, is_return_log_prob::Bool=false)
+    # Omit specific code
+    if is_sampling
+        if is_return_log_prob
+            return tanh.(z), logp_π
+        else
+            return tanh.(z)
+        end
+    else
+        return μ, logσ
+    end
+end
+### Inference
+function (model::GaussianNetwork)(state, action)
+    # Omit specific code
+    return logp_π
+end
+```
+
+##### Variational Auto-Encoder (VAE)
+In offline reinforcement learning tasks, VAE\dcite{DBLP:journals/corr/KingmaW13} is often used to learn from datasets to approximate behavior policy.
+
+VAE we implemented consists of two neural network: `encoder` and `decoder` ([link](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/src/ReinforcementLearningCore/src/policies/q_based_policies/learners/approximators/neural_network_approximator.jl)).
+```julia
+Base.@kwdef struct VAE{E, D}
+    encoder::E
+    decoder::D
+end
+```
+In the encoding stage, it accepts input state and action and outputs the mean and standard deviation of the distribution. Afterward, the hidden action is obtained by sampling the distribution. In the decoding stage, state and hidden action are used as the input to reconstruct action.
+
+During training, we call the `vae_loss` function to get the reconstruction loss and KL loss. The specific task determines the ratio of these two losses.
+
+```julia
+function vae_loss(model::VAE, state, action)
+    # Omit specific code
+    return recon_loss, kl_loss
+end
+```
+
+In the specific algorithm, the functions that may need to be called are as follows:
+
+```julia
+### Encode + decode
+function (model::VAE)(state, action)
+    ### Omit specific code
+    return a, μ, σ
+end
+### Decode
+function decode(model::VAE, state, z)
+    ### Omit specific code
+    return a
+end
+```
+
+#### Offline RL Algorithms
+We used the existing algorithms and hooks to train the offline RL algorithm to create datasets in several environments (such as CartPole, Pendulum) for training. This work can guide the subsequent development of package RLDataset.jl, for example:
+```julia
+gen_dataset("JuliaRL-CartPole-DQN", policy, env)
+```
+
+##### Benchmark
+We implemented and experimented with offline DQN (in discrete action space) and offline SAC (in continuous action space) as benchmarks. The performance of offline DQN in Cartpole environment:
```

Comment:
The tense you used in this report is kind of confusing to me. Sometimes the present tense is used, now the past tense here. Better to unify them all.